Documentation/proposals/202407-remote-write.md

```diff
@@ -0,0 +1,121 @@
+# RemoteWrite CRD
+
+- Owners:
+  - [@superbrothers](https://github.com/superbrothers)
+- Related Tickets:
+  - [#6508](https://github.com/prometheus-operator/prometheus-operator/issues/6508)
+- Other docs:
+  - n/a
+
+## TL;DR
+
+This design doc proposes RemoteWrite CRD, which enables cluster admins to delegate the ability to configure Prometheus remote_rewrite configuration to application developers/operators.
+
+## Why
+
+The Prometheus remote_write configuration is defined in the Prometheus CRD. Currently, the configuration data generation is the responsibility of cluster admins.
+
+## Goals
+
+The main goal is for application developers/operators to configure themselves how the remote endpoint sends metrics.
+
+This means exposing a new CRD to configure Prometheus remote_write configuration.
+
+## Non-goals
+
+We do not cover "RemoteRead" CRD here. We can still implement it later if needed.
+
+## How
+
+### RemoteWrite CRD
+
+The RemoteWrite CRD represents one of the Prometheus remote_write configuration scoped to the resource’s namespace.
+
+```yaml
+apiVersion: monitoring.coreos.com/v1alpha1
+kind: RemoteWrite
+metadata:
+  name: example
+  namespace: default
+spec:
+  url: "https://aps-workspaces.us-west-2.amazonaws.com/workspaces/<workspace id>/api/v1/remote_write"
+  sigv4:
+    region: us-west-2
+    accessKey:
+      name: aws_access
+      key: access_key
+    secretKey:
+      name: aws_access
+      key: secret_key
+```
+
+```go
+package v1alpha1
+
+import (
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type RemoteWrite struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec monitoringv1.RemoteWriteSpec `json:"spec"`
+}
+```
+
+### Prometheus CRD
+
+The Prometheus CRD is extended with 2 new fields (remoteWriteSelector and remoteWriteNamespaceSelector) that define which RemoteWrite resources are associated with this Prometheus instance.
+
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: Prometheus
+metadata:
+  name: example
+  namespace: default
+spec:
+  # RemoteWrite objects to be selected. If nil, nothing will be selected.
+  remoteWriteSelector: {}
+  # Namespace(s) to be selected for RemoteWrite discovery. If nil, only check in its own namespace.
+  remoteWriteNamespaceSelector: {}
+  ...
+```
+
+#### Respect enforceNamespaceLabel and excludedFromEnforcement
+
+The RemoteWrite CRD should respect the Prometheus `.spec.enforcedNamespaceLabel` field.
+
+The object’s namespace is added as the last item in the `write_relabel_config` of the remote_write configuration.
+
+```yaml
+write_relabel_configs:
+...
+- target_label: <enforced-namespace-label>
+  replacement: <namespace>
+```
+
+The RemoteWrite CRD should also respect the Prometheus `.spec.excludedFromEnforcement` field.
```

Comment:
Side-note: the new CRD would need to be added to the list of supported resources.